src/dynamic-element.js

```diff
@@ -1,15 +1,40 @@
 /*eslint padded-blocks:0*/
-import {useView, customElement, bindable} from 'aurelia-templating';
-export function _createDynamicElement(name: string, viewUrl: string, bindableNames: string[]): Function {
+import {useView, customElement, bindable, useShadowDOM} from 'aurelia-templating';
+export function _createDynamicElement({ name, viewUrl, bindableNames, useShadowDOMmode } : {
+  name: string,
+  viewUrl: string,
+  bindableNames: string[],
+  useShadowDOMmode: null | '' | 'open' | 'closed'
+}): Function {
   @customElement(name)
   @useView(viewUrl)
   class DynamicElement {
     bind(bindingContext) {
       this.$parent = bindingContext;
     }
   }
+
   for (let i = 0, ii = bindableNames.length; i < ii; ++i) {
     bindable(bindableNames[i])(DynamicElement);
   }
+
+  switch (useShadowDOMmode) {
+  case 'open':
+    useShadowDOM({ mode: 'open' })(DynamicElement);
+    break;
+
+  case 'closed':
+    useShadowDOM({ mode: 'closed' })(DynamicElement);
+    break;
+
+  case '':
+    useShadowDOM(DynamicElement);
+    break;
+
+  default:
+    // Do not use shadow dom
```

Comment:
I think there should be a warning here
```js
import { getLogger } from 'aurelia-logging';

    // Do not use shadow dom
    getLogger('aurelia-html-only-element')
      .warn(`Expected 'use-shadow-dom' value to be "close", "open" or "", received ${useShadowDOMmode}`)
```

Reply:
ok I could do that. I will also have to include a undefined case that does nothing to be able to default to a warning.

Reply:
Added warning + null case when attribute is not set